Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp

```diff
@@ -863,13 +863,14 @@ class HTMLFastPathParser {
         return WTFMove(element);
     }
 
-    template<typename HTMLElementType> Ref<HTMLElementType> parseVoidElement(Ref<HTMLElementType>&& element)
+    template<typename HTMLElementType> Ref<HTMLElementType> parseVoidElement(Ref<HTMLElementType>&& element, ContainerNode& parent)
     {
         parseAttributes(element);
         if (!parsingFailed()) {
             element->beginParsingChildren();
             element->finishParsingChildren();
         }
+        parent.parserAppendChild(element);
```

Comment:
There is a slight bug here. We must not call `parent.parserAppendChild` when parsing had failed. This is why some tests are hitting debug assertions.